consul/client.go

```diff
@@ -328,50 +321,34 @@ func (c *Client) localEvent(event serf.UserEvent) {
 
 // RPC is used to forward an RPC call to a consul server, or fail if no servers
 func (c *Client) RPC(method string, args interface{}, reply interface{}) error {
-	// Check the last rpc time
-	var server *serverParts
-	if time.Now().Sub(c.lastRPCTime) < clientRPCCache {
-		server = c.lastServer
-		if server != nil {
-			goto TRY_RPC
-		}
-	}
-
-	// Bail if we can't find any servers
-	c.consulLock.RLock()
-	if len(c.consuls) == 0 {
-		c.consulLock.RUnlock()
+	server := c.serverMgr.FindServer()
+	if server == nil {
+		c.logger.Printf("[ERR] consul: No healthy servers found in the server config")
```

Comment:
I'm going to do the inverse, keep the log message in `FindServer()` only.  In FindServer that is a valid warning.  RPC returns error and leaves it to the higher level caller to do whatever it wants.  As an operator, I have the detail necessary to know why the RPC call(s) failed.  The new wording is pretty straightforward: `[WARN] consul: No servers available`